### ISSUE TITLE
MapCanvasScrollControllerMixin:GetCursorPosition should account for the map scale

### ISSUE THREAD

**Issue by Nevcairiel** [Stanzilla/WoWUIBugs#338]
`MapCanvasScrollControllerMixin:GetCursorPosition` is used in the World Map to compute where on the map the user has clicked for various interactions. To accurately determine this position, the scale of the map frame needs to be taken into account. Unfortunately, only the scale of UIParent is currently taken into account, not of the map frame itself (because the default UI does not allow you to rescale it)

A simple addon to change the scale of the map to better fit the players desire for the size of the map is therefor rather complicated, and in 10.0 with taint spreading through Edit Mode also quite dangerous.

A much simpler way would be if `MapCanvasScrollControllerMixin:GetCursorPosition` would simply account for it - to no detriment of the default UI, and a very simple change:

```diff
--- a/Interface/AddOns/Blizzard_MapCanvas/MapCanvas_ScrollContainerMixin.lua
+++ b/Interface/AddOns/Blizzard_MapCanvas/MapCanvas_ScrollContainerMixin.lua
@@ -745,7 +745,7 @@ end

 function MapCanvasScrollControllerMixin:GetCursorPosition()
        local currentX, currentY = GetCursorPosition();
-       local effectiveScale = UIParent:GetEffectiveScale();
+       local effectiveScale = self:GetMap():GetEffectiveScale();
        return currentX / effectiveScale, currentY / effectiveScale;
 end
```

With this change, the full scale of the map frame is being taken into account, and the scale of eg. the WorldMapFrame can easily be changed without mouse interactions hiting the wrong parts of the map - and most importantly, without needing to taint the inner workings of the map with complex hooks into all the calculations.

This logic is overall more accurate as well, if the default UI ever decides to give scale control over the map into EditMode, or anything like that.

**Comment by Meorawr** [comment]
This should hopefully be fixed in an upcoming 10.0.5 PTR build.

**Comment by Meorawr** [comment]
Fixed in 10.0.5.47481.
